Engine/src/Engine/Scene.cpp

```diff
@@ -335,66 +380,71 @@ namespace Engine
 #pragma region Renderable Entities
 	
 	//Return the VBO for sprites. If it doesn't exist, create it.
-	GLuint Scene::getSpriteVBO() 
+	GLuint Scene::getVBO(std::string message) 
 	{
-		if(!m_createdVBO);
-		{
-			m_createdVBO = true;
+		GLuint vbo;
+		glGenBuffers(1, &vbo);
+		glBindBuffer(GL_ARRAY_BUFFER, vbo);
 
+		if (message == "sprite") 
+		{
 			float vertices[] = 
 			{
 				// positions  // texture coords (UV coords)
-
 				0.f, 0.f, 0.f,  0.f, 0.f,  // top left
 				1.f, 0.f, 0.f,  1.f, 0.f,  // top right
 				1.f, 1.f, 0.f,  1.f, 1.f,  // bottom right
 				0.f, 1.f, 0.f,  0.f, 1.f,  // bottom left
 			};
-			
-			
-			glGenBuffers(1, &m_spriteVBO);
-			glBindBuffer(GL_ARRAY_BUFFER, m_spriteVBO);
-
 			glBufferData(GL_ARRAY_BUFFER, sizeof(vertices), vertices, GL_DYNAMIC_DRAW);
 		}
-
-		return m_spriteVBO;
+		else if (message == "rectangle") 
+		{
+			float vertices[] = 
+			{
+				// positions 
+				0.f, 0.f, 0.f,  // top left
+				1.f, 0.f, 0.f,  // top right
+				1.f, 1.f, 0.f,  // bottom right
+				0.f, 1.f, 0.f   // bottom left
+			};
+			glBufferData(GL_ARRAY_BUFFER, sizeof(vertices), vertices, GL_DYNAMIC_DRAW);
+		}
+		else 
+		{
+			GE_CORE_ERROR("A vbo was declared with incorrect vertex data.");
+		}
+		return vbo;
 	}
 
 	//Return the VAO for sprites. If it doesn't exist, create it.
-	GLuint Scene::getSpriteVAO()
+	GLuint Scene::getVAO()
 	{
-		if (!m_createdVAO)
-		{
-			m_createdVAO = true;
-
-			glGenVertexArrays(1, &m_spriteVAO);
-			glBindVertexArray(m_spriteVAO);
-		}
-		
-		return m_spriteVAO;
+		GLuint vao;
+		glGenVertexArrays(1, &vao);
+		glBindVertexArray(vao);
+		return vao;
 	}
 
 	//Return the IBO for sprites. If it doesn't exist, create it.
-	GLuint Scene::getSpriteIBO()
+	GLuint Scene::getIBO(std::string message)
 	{
-		if (!m_createdIBO);
-		{
-			m_createdIBO = true;
-			
+		GLuint ibo;
+		glGenBuffers(1, &ibo);
+		glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, ibo);
+
+		if (message == "rectangle") {
```

Comment:
If we're going to have a 'sprite' and a 'rectangle' we should probably declare them separately. This comes back to the question of "why do we want two objects". 

Ultimately, if we are going to use the same indices/vertices for an object, we should store them in a variable (map or otherwise) to access the same data and not re-declare it each time, IMO.

Reply:
Great observation. I came to this conclusion because when I initially had static variables set that can be reused (rectangle or sprite), OpenGL seems to in some way operate directly with the set of vertices and in result, only the last object gets drawn. So the only conclusion I could arrive to is redeclare the vertices per buffer object.

Reply:
I think that was just based on how we were previously storing the vertices. If we store them in separate VBOs etc, it should fix but I'm not 100% sure on that. Worth exploring though, for sure.